src/lib/components/VideoCard/VideoCard.svelte

```diff
@@ -1,78 +1,151 @@
 <script lang="ts">
   import "./VideoCard.scss";
 
-  import { afterUpdate } from "svelte";
-
-  import getYoutubeVideoData from "$lib/services/youtube";
+  import type { Thumbnails } from "$lib/services/server/youtube";
+  import getYoutubeVideoIDFromURL from "$lib/util/getYoutubeVideoIDFromURL";
   import YoutubeSubscribeLink from "$lib/components/YoutubeSubscribeLink";
+  import Image, { type Sources, type Srcset } from "$lib/components/Image";
 
   type VideoCardVariation = "hero" | "primary" | "secondary" | "tertiary";
   interface $$Props {
     class?: string;
     url: string;
-    customTitle?: string | null | undefined;
-    customDescription?: string | null | undefined;
+    title?: string | null | undefined;
+    description?: string | null | undefined;
     variation?: VideoCardVariation;
+    blurhash?: string | null | undefined;
+    thumbnails?: Thumbnails;
+    sizeType?: SizeType;
   }
 
   let className = "";
   export { className as class };
   export let url = "";
-  export let customTitle: $$Props["customTitle"] = null;
-  export let customDescription: $$Props["customDescription"] = null;
+  export let title: $$Props["title"] = null;
+  export let description: $$Props["description"] = null;
   export let variation = "primary";
 
-  let youtubeVideoId: string | null;
+  const maxDescriptionLength = 500;
+  $: trimmedDescription = description?.slice(0, maxDescriptionLength);
```

Comment:
I'm fine with setting a maximum length, but we should still cut the description off on newlines to avoid some of the issues I was seeing during development (e.g. showing hashtags that don't do anything, URLs going too long and squishing the video, etc.).